Sources/LanguageServerProtocol/Types/TextDocument.swift

```diff
@@ -74,6 +56,57 @@ extension TextDocument {
         return TextDocument(uri: uri, languageId: languageId, version: version, text: andText)
     }
 
+    /// Converts the position to a zero-based byte offset in the TextDocument.
+    ///
+    /// - Parameter position: The position to convert.
+    /// - Returns: The byte offset from the TextDocument start index.
+    /// - Throws: WorkspaceError.positionNotFound if the Position is not within the bounds of the TextDocument.
+    func byteOffset(at position: Position) throws -> Int {
+        let seq = AnySequence { LineIterator(self.text) }
+        guard let line = seq.first(where: { $0.number == position.line }) else {
+            throw WorkspaceError.positionNotFound
+        }
+        let limit = (line.last) ? text.endIndex : text.index(before: line.end)
+        guard let final = text.index(line.start, offsetBy: position.character, limitedBy: limit) else {
+            throw WorkspaceError.positionNotFound
+        }
+        return text.distance(from: text.startIndex, to: final)
+    }
+
+    /// Create a Position in a TextDocument from a byte offset.
+    ///
+    /// - Parameter offset: The offset from the start of the TextDocument.
+    /// - Returns: A new and valid Position.
+    /// - Throws: WorkspaceError.positionNotFound if the offset is not within the bounds of the TextDocument.
+    func position(for offset: Int) throws -> Position {
```

Comment:
Does this make more sense as a function on `TextDocument` _or_ as an initializer on `Position` that takes a `TextDocument` and offset? 🤔